### PR TITLE
add smoothap

### PR DESCRIPTION
**Pull request by interestingzhuo**
add smoothap loss (ECCV2020)

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thanks for the pull request. Loss functions should extend BaseMetricLossFunction. See [TripletMarginLoss](https://github.com/KevinMusgrave/pytorch-metric-learning/blob/master/src/pytorch_metric_learning/losses/triplet_margin_loss.py) for an example.

**Comment by KevinMusgrave** [comment]
Closing due to inactivity
